### PR TITLE
Fix codefactor issues

### PR DESCRIPTION
**Pull request by JoranAngevaare**
# Cleaner code to shutup codefactor
An attempt to set this repo to 0 issues, also see #831 

## PR contents:
 - remove TODO's especially since most are (soon) addressed in CMT2 or [simpler RunDB ](https://github.com/XENONnT/straxen/pull/591) logic that is still pending
 - Remove duplicated code
 - Remove unused shell scripts

### CODE REVIEW

File: straxen/plugins/event_patternfit.py [hunk 1]
```diff
@@ -415,28 +415,36 @@ def lbinom_pmf_diriv(k, n, p, dk=1e-7):
...
-        x = max(x, x_min)
-
-        dx = abs(x - x1)
-        x0 = x1
-        x1 = x
-
+        y0 = lbinom_pmf(x0, *args)
+        y1 = lbinom_pmf(x1, *args)
+        dx, x0, x1 = _numeric_derivative(y0, y1, err, target, x_min, x_max, x0, x1)
         max_iter -= 1
```
Comment (JoranAngevaare) [inline straxen/plugins/event_patternfit.py:447]: @GiovanniVolta , would you be able to quickly cross check that this change (that reduces code duplication where codefactor was complaining about) doesn't affect the code. Unless I made a typo, performance and outcome should be exactly the same.

File: straxen/plugins/event_patternfit.py [hunk 2]
```diff
@@ -415,28 +415,36 @@ def lbinom_pmf_diriv(k, n, p, dk=1e-7):
...
-        x = (target - y0) / (y1 - y0) * (x1 - x0) + x0
-        x = min(x, x_max)
-        x = max(x, x_min)
-
-        dx = abs(x - x1)
-        x0 = x1
-        x1 = x
-
+        y0 = lbinom_pmf(x0, *args)
+        y1 = lbinom_pmf(x1, *args)
```
Comment (WenzDaniel) [inline straxen/plugins/event_patternfit.py:445]: These two should be `lbinom_pmf_diriv` is not it? Also tagging @GiovanniVolta for final approval.

File: straxen/plugins/pulse_processing.py [hunk 3]
```diff
@@ -185,7 +185,7 @@ def compute(self, raw_records, start, end):
 
         # Do not trust in DAQ + strax.baseline to leave the
         # out-of-bounds samples to zero.
-        # TODO: better to throw an error if something is nonzero
+        # maybe better to throw an error if something is nonzero
```
Comment (WenzDaniel) [inline straxen/plugins/pulse_processing.py:188]: Any intention to change this? I think we can remove the comment entirely.

**Review by WenzDaniel — COMMENTED** [review]
Just one bug I think also tagged Giovanni.

File: straxen/plugins/pulse_processing.py (hunk 3 above)
Comment (JoranAngevaare) [reply straxen/plugins/pulse_processing.py:188]: sure

File: straxen/plugins/event_patternfit.py (hunk 2 above)
Comment (JoranAngevaare) [reply straxen/plugins/event_patternfit.py:445]: ay, so many versions floating around that I missed this, thanks yes you are right!

File: straxen/analyses/event_display.py [hunk 4]
```diff
@@ -39,21 +39,13 @@
...
-                         records_matrix=True,
-                         s2_fuzz=50,
-                         s1_fuzz=0,
-                         max_peaks=500,
-                         xenon1t=False,
+def event_display_simple(context, run_id, events, to_pe,
+                         records_matrix=True, s2_fuzz=50, s1_fuzz=0,
+                         max_peaks=500, xenon1t=False, s1_hp_kwargs=None,
+                         s2_hp_kwargs=None, event_time_limit=None,
+                         plot_all_positions=True,
```
[the remark this answers is not in the document]
Comment (JoranAngevaare) [reply straxen/analyses/event_display.py]: Okay, you are right, probably a better way of avoiding duplicate code is to not duplicate it in the first place. I've deprecated `event_display_simple` and just made it a simple option of the `event_display`

File: straxen/plugins/event_patternfit.py (hunk 2 above)
Comment (GiovanniVolta) [reply straxen/plugins/event_patternfit.py:445]: Yes, Daniel is right. Here we need the derivative in order to find x such that der(x0)<= der(x) <= der(x1)
```
y0 = lbinom_pmf_diriv(x0, *args) 
y1 = lbinom_pmf_diriv(x1, *args)
```

File: straxen/plugins/event_patternfit.py [hunk 5]
```diff
@@ -448,20 +456,10 @@ def lbinom_pmf_inverse(x_min, x_max, target, args, err=1e-7, max_iter=50):
...
-        x = (target - y0) / (y1 - y0) * (x1 - x0) + x0
-        x = min(x, x_max)
-        x = max(x, x_min)
-
-        dx = abs(x - x1)
-        x0 = x1
-        x1 = x
-
+        y0 = lbinom_pmf_diriv(x0, *args)
+        y1 = lbinom_pmf_diriv(x1, *args)
```
Comment (GiovanniVolta) [inline straxen/plugins/event_patternfit.py:460]: Here we need **lbinom_pmf**

**Review by GiovanniVolta — CHANGES_REQUESTED** [review]

**Comment by JoranAngevaare** [comment]
@GiovanniVolta thanks! For having a quick look (🤦I messed up) , is it better now?

**Review by GiovanniVolta — APPROVED** [review]
Thank you @JoranAngevaare! I have tested the binomial test, and it looks good to me. Maybe @zhut19 wants a have a quick look as well?

**Review by WenzDaniel — APPROVED** [review]

**Pull request merged**